### PR TITLE
MDEV-29387 Period name with more than 32 symbols crashes the server

### PR DESCRIPTION
**Pull request by Weijun-H**
Signed-off-by: Weijun-H <huangweijun1001@gmail.com>

- [x] *The Jira issue number for this PR is: MDEV-29387*

## Description

The following query:
```create table t2 (s date, e date, period for `asdf123456789012345678901234567890` (s,e));```
lead to crash.
I increased `MAX_FIELD_NAME`  to avoid it.

## How can this PR be tested?
Adds MTR cases.

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [x] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

### CODE REVIEW

File: sql/sql_const.h [hunk 1]
```diff
@@ -29,7 +29,7 @@
 /* extra 4+4 bytes for slave tmp tables */
 #define MAX_DBKEY_LENGTH (NAME_LEN*2+1+1+4+4)
 #define MAX_ALIAS_NAME 256
-#define MAX_FIELD_NAME 34			/* Max colum name length +2 */
+#define MAX_FIELD_NAME (NAME_CHAR_LEN+2)			/* Max colum name length +2 */
```
Comment (FooBarrior) [inline sql/sql_const.h:32]: Why you used NAME_CHAR_LEN? I see that NAME_LEN is used in most places for an array allocation (one extra byte, plus one for zero termination). So i'd suggest NAME_LEN+1, as MAX_FIELD_NAME is used for an array size without +1.

**Review by FooBarrior — CHANGES_REQUESTED** [review]
Hello @Weijun-H! Thanks for a contribution.

I like the simplicity of the patch. Please address my comments before we can proceed to merging

File: mysql-test/main/period_name.test [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--echo #
```
[the remark this answers is not in the document]
Comment (Weijun-H) [reply mysql-test/main/period_name.test:1]: I did not find the relative test. Which version did you mention?

File: sql/sql_const.h (hunk 1 above)
Comment (vuvova) [reply sql/sql_const.h:32]: `NAME_CHAR_LEN` is the max length in *characters*, `NAME_LEN` is the max length in *bytes*.

`MAX_FIELD_NAME` is in bytes. So it must be `NAME_LEN`, `NAME_CHAR_LEN` is simply defined in different units.

Comment (FooBarrior) [reply sql/sql_const.h:32]: Agree, so it should be `(NAME_LEN + 1)`

File: mysql-test/main/period_name.test (hunk 2 above)
[the remark this answers is not in the document]
Comment (FooBarrior) [reply mysql-test/main/period_name.test:1]: It's in all versions that feature application-time periods. See `mysql-test/suite/period`.

File: sql/sql_const.h [hunk 3]
```diff
@@ -29,7 +29,7 @@
 /* extra 4+4 bytes for slave tmp tables */
 #define MAX_DBKEY_LENGTH (NAME_LEN*2+1+1+4+4)
 #define MAX_ALIAS_NAME 256
-#define MAX_FIELD_NAME 34			/* Max colum name length +2 */
+#define MAX_FIELD_NAME (NAME_LEN+1)			/* Max colum name length +2 */
```
Comment (FooBarrior) [inline sql/sql_const.h:32]: now you need to adjust the comment column  and fix its content to +1

File: sql/sql_const.h [hunk 4]
```diff
@@ -29,7 +29,7 @@
 /* extra 4+4 bytes for slave tmp tables */
 #define MAX_DBKEY_LENGTH (NAME_LEN*2+1+1+4+4)
 #define MAX_ALIAS_NAME 256
-#define MAX_FIELD_NAME 34			/* Max colum name length +2 */
+#define MAX_FIELD_NAME (NAME_LEN+1)			/* Max colum name length +1 */
```
Comment (FooBarrior) [inline sql/sql_const.h:32]: Oh, i see what was the problem in markup now: there are tabs used in between. So:
1. Note that tabs are configured at width 8 for mariadb.
2. Please use spaces here instead. The comment should be aligned with other neghbouring comments

File: mysql-test/suite/period/t/create.test [hunk 5]
```diff
@@ -85,3 +85,14 @@ insert t values (2, '2001-01-01', '2001-01-01');
...
+
+--echo # MDEV-29387: Period name with more than 32 symbols crashes the server
+--echo #
+
+--echo # test 34 symbols
+create table t2 (s date, e date, period for `abcd123456789012345678901234567890` (s,e)); 
+drop table t2;
+
+--echo # test 64 symbols
+create table t2 (s date, e date, period for `abcd123456789012345678901234567890123456789012345678901234567890` (s,e));
```
Comment (FooBarrior) [inline mysql-test/suite/period/t/create.test:97]: An 80 symbols width should be maintained for handwritten files.

**Review by FooBarrior — CHANGES_REQUESTED** [review]
@Weijun-H thanks for rapid fixing. I guess you could have had tab width misconfigured,
but the comment is still not aligned. You can check it in github changes.

I also noticed that a `create table` for 64-sym periond overflows our 80-sym width rule. Please fix it as well.

File: mysql-test/suite/period/t/create.test [hunk 6]
```diff
@@ -90,9 +90,12 @@ drop table t;
...
+create table t2 (s date, e date, 
+                 period for `abcd123456789012345678901234567890` (s,e)); 
 drop table t2;
 
 --echo # test 64 symbols
-create table t2 (s date, e date, period for `abcd123456789012345678901234567890123456789012345678901234567890` (s,e)); 
+create table t2 (s date, e date, period for 
+                 `abcd123456789012345678901234567890123456789012345678901234567890`
+                 (s,e)); 
 drop table t2;
```
Comment (FooBarrior) [inline mysql-test/suite/period/t/create.test:101]: please also add a line feed iin the end of this file

**Review by FooBarrior — CHANGES_REQUESTED** [review]

**Review by FooBarrior — APPROVED** [review]

**Pull request merged**